main/src/test/java/org/mobilitydata/gtfsvalidator/validator/OverlappingFrequencyValidatorTest.java

```diff
@@ -0,0 +1,114 @@
+                .setHeadwaySecs(headwaySecs).build();
+    }
+
+    private List<Notice> validateFrequencies(GtfsFrequency... frequencies) {
+        NoticeContainer noticeContainer = new NoticeContainer();
+        OverlappingFrequencyValidator validator = new OverlappingFrequencyValidator();
+        validator.table = GtfsFrequencyTableContainer.forEntities(Arrays.asList(frequencies), noticeContainer);
+        validator.validate(noticeContainer);
+        return noticeContainer.getNotices();
+    }
+
+    @Test
+    public void validSequentialInOrder() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "07:00:00", "10:00:00", 300)))
+                .isEmpty();
+    }
+
+    @Test
+    public void validSequentialReversed() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "07:00:00", "10:00:00", 300),
+                createFrequency(3, "t0", "05:00:00", "07:00:00", 600)))
+                .isEmpty();
+    }
+
+    @Test
+    public void validWithGap() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "08:00:00", "10:00:00", 300)))
+                .isEmpty();
+    }
+
+    @Test
+    public void validDifferentTrips() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t1", "06:00:00", "10:00:00", 300)))
+                .isEmpty();
+    }
+
+    @Test
+    public void overlappingPartially() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "06:00:00", "10:00:00", 300)))
+                .containsExactly(new OverlappingFrequencyNotice(
+                        2, GtfsTime.fromString("07:00:00"),
+                        3, GtfsTime.fromString("06:00:00"), "t0"));
+    }
+
+    @Test
+    public void overlappingIncludedSameStart() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "05:00:00", "06:30:00", 300)))
+                .containsExactly(new OverlappingFrequencyNotice(
+                        3, GtfsTime.fromString("06:30:00"),
+                        2, GtfsTime.fromString("05:00:00"), "t0"));
+    }
+
+    @Test
+    public void overlappingIncludedSameEnd() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "06:30:00", "07:00:00", 300)))
+                .containsExactly(new OverlappingFrequencyNotice(
+                        2, GtfsTime.fromString("07:00:00"),
+                        3, GtfsTime.fromString("06:30:00"), "t0"));
+    }
+
```

Comment:
I would suggest adding a test case for when one trip is fully contained within the other and doesn't share start or end times, which should generate a notice:

~~~
     * a |--------|            07:00 am to 12:00 am
     * b   |----|              08:00 am to 11:00 am
~~~

Reply:
Done.